### ISSUE TITLE
GTK4 Invalid property: GtkBox.margin_left

### ISSUE THREAD

**Issue by 05storm26** [neffo/bing-wallpaper-gnome-extension#98]
```
Gtk.BuilderError: /usr/share/gnome-shell/extensions/BingWallpaper@ineffable-gmail.com/Settings.ui:53:39 Invalid property: GtkBox.margin_left

Stack trace:
  buildPrefsWidget@/usr/share/gnome-shell/extensions/BingWallpaper@ineffable-gmail.com/prefs.js:38:15
  _init@resource:///org/gnome/Shell/Extensions/js/extensionsService.js:209:40
  OpenExtensionPrefsAsync/<@resource:///org/gnome/Shell/Extensions/js/extensionsService.js:122:28
  asyncCallback@resource:///org/gnome/gjs/modules/core/overrides/Gio.js:115:22
  run@resource:///org/gnome/Shell/Extensions/js/dbusService.js:177:20
  main@resource:///org/gnome/Shell/Extensions/js/main.js:19:13
  run@resource:///org/gnome/gjs/modules/script/package.js:206:19
  start@resource:///org/gnome/gjs/modules/script/package.js:190:8
  @/usr/share/gnome-shell/org.gnome.Shell.Extensions:1:17

**Comment by 05storm26** [comment]
I see that there is a separate Settings4.ui xml for Gtk4. I wonder why it is trying to use the old one for me?

**Comment by 05storm26** [comment]
The 30th release on the release page is from the commit: 467942656135ae5339169c903f5b0cc80ac5e6ca (https://github.com/neffo/bing-wallpaper-gnome-extension/commit/467942656135ae5339169c903f5b0cc80ac5e6ca). 

This has no gnome 40 support yet. 

@neffo The version 30th description says that it supports gnome 40, is it built from the wrong commit?

**Comment by 05storm26** [comment]
There is also a v29a release, which is newer than v30

**Comment by neffo** [comment]
Are you running the latest release from extensions.gnome.org or the git from here? I presume you are running Gnome 40?

GNOME allows you to disable version checking with extensions, do you have that setting enabled?

![image](https://user-images.githubusercontent.com/5569112/115133975-b7b6c080-a04f-11eb-8d51-6bb87ebce88b.png)

(Don't worry about the github version tags, I'll fix that up.)

**Comment by neffo** [comment]
v30 on extensions.gnome.org added GNOME 40 support, [see diff here](https://extensions.gnome.org/review/23778), I'd consider the version numbers there the _canonical_ versions, with git it's better to refer to commit. The versions got mangled and the github UI doesn't easily let you fix that, so sorry for the confusion.

**Comment by 05storm26** [comment]
@neffo  Okay can you create a new release maybe? Like v31?

**Comment by neffo** [comment]
Yeah, I'll roll up a new release soon (when some fixes get implemented). Not sure it makes sense to create a new release to fix an issue with the Github releases. They can be fixed with git commands, but not via the Github UI (apparently).

I suspect I'm not the only one but I'm a bit skittish on running git commands with --delete tags in them if I don't understand them completely.

**Comment by 05storm26** [comment]
It is not fixed

**Comment by neffo** [comment]
What is not fixed? Have you downloaded the version from extensions.gnome.org?

**Comment by 05storm26** [comment]
Sorry I thought by this:

> EDIT: hopefully fixed

you meant that the github releases page is fixed for v30. Apparently not. 

> Have you downloaded the version from extensions.gnome.org?
I have not tried it since I don't like to install stuff using anything other than the OS package manager for which the package was using the github release page. Anyway I was able to get around that: https://aur.archlinux.org/packages/gnome-shell-extension-bing-wallpaper/#comment-804724

Also now there is a v31 so this issue is now fixed. Thanks anyway!
